rfc/egress-traffic.md

```diff
@@ -0,0 +1,238 @@
+
+### Example Pricing Tiers:
+
+1. **Egress Starter:**
+
+   - **0 to 5 GB:** Free
+   - **5 GB to 20 GB:** $0.10 per GB
+   - **20 GB+:** $0.07 per GB
+
+2. **Egress Lite:**
+
+   - **0 to 10 GB:** Free
+   - **10 GB to 50 GB:** $0.08 per GB
+   - **50 GB+:** $0.06 per GB
+
+3. **Egress Business:**
+   - **0 to 20 GB:** Free
+   - **20 GB to 100 GB:** $0.05 per GB
+   - **100 GB+:** $0.03 per GB
+
+### Possible Solution
+
+Since **Usage Records** are being deprecated by Stripe and **Billing Meters** offer greater flexibility and scalability, adopting **Billing Meters** for Storacha Network’s usage-based billing is recommended.
+
+### Benefits of Billing Meters:
+
+- **No Customer Disruption:** Existing billing remains intact while usage is tracked in the background through the Billing Meter.
+- **Flexible Tiers:** New prices can be created for each product tier and associated with the Billing Meter for automated usage tracking.
+- **Scalability:** This approach ensures that billing for egress traffic is accurate and scalable with minimal operational overhead.
+
+## Questions & Answers
+
+### 1. **Operational Complexity:**
+
+- **Question:** Which approach requires less effort to maintain in the long run? Can the additional infrastructure of Alternative 1 or 3 be justified given their decoupled architecture?
+- **Answer:** Alternative 2 requires less effort to maintain in the long run since it leverages existing infrastructure (`w3up/upload-api`), avoiding the need to set up new infrastructure components (SQS, Lambda, DynamoDB).
+
+### 2. **Security:**
+
+- **Question:** How should delegation and authentication mechanisms be handled, particularly in Alternative 2? What security implications arise from allowing the Cloudflare Freeway Worker to directly interact with the `w3up/upload-api`?
+- **Answer:** Delegation can be handled by granting the Cloudflare Freeway Worker permission to invoke the required capabilities. The Gateway will need to become a `w3up` client to claim the delegation and execute the invocation securely.
+- **Question:** Who will create the delegation to the Cloudflare Freeway Worker?
+- **Answer:** ???
+
+### 3. **Performance Impact:**
+
+- **Question:** Will the real-time nature of API updates in Alternative 2 introduce latency concerns for the Cloudflare Freeway Worker? How does this compare to the event-driven model in Alternatives 1 and 3?
+- **Answer:** By using Cloudflare’s "fire and forget" context calls, the Freeway Worker can avoid blocking while serving the resource. This minimizes latency, allowing the request to complete without waiting for the billing events, making the performance impact negligible.
+
+### 4. **Logging and Monitoring:**
+
+- **Question:** How can logs and monitoring be effectively implemented in both alternatives? Does Alternative 2’s reliance on signed receipts offer sufficient logging capabilities? Do we really need these signed receipts?
+
+- **Answer:** ???
```

Comment:
I do think it's at least moderately important to have those signed receipts - I suspect customers are going to want as much information as possible about what they're being charged for. We might be able to just aggregate metrics around them and store in Glacier for the sake of minimizing costs, but having signed receipts for each request feels like something that will eventually be fairly useful?

@alanshaw is right that there will be a LOT of them and they'll probably have different retention policies than most other UCAN invocation receipts, though for the sake of getting this shipped we might want to use the upload API service for now and add a new service later?

It definitely might be worth adding a new service that uses a similar invocation processing pipeline and different retention policies - ideally we'd move egress tracking receipts to Glacier pretty quickly.